conftest.py

```diff
@@ -244,7 +247,12 @@ def get_test_launcher() -> str:
 
     @staticmethod
     def get_test_port() -> int:
-        return test_port
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+            for port in range(test_port, test_port + test_num_ports):
+                result = sock.connect_ex(("127.0.0.1", port))
+                if result:
+                    return port
```

Comment:
I'm a little confused on the logic here: It looks like the idea is that this socket is going to try to connect to this addr/port, and if there is some error we assume that it is because there is no socket bound there to accept the connection, and therefore the port must be free.

However, errors could come from a variety of problems (e.g. there is a socket bound but it is not listening, there is a socket bound and it is listening but it ran out of buffer space for pending new unaccepted connections, etc.) where the port may not actually be available.

Should this be trying to bind instead??

Reply:
Switched to `bind` to avoid port conflicts